readthedocs/oauth/services/githubapp.py

```diff
@@ -0,0 +1,473 @@
+            if remote_repo:
+                remote_repositories.append(remote_repo)
+
+        repos_to_delete = self.installation.repositories.exclude(
+            pk__in=[repo.pk for repo in remote_repositories],
+        ).values_list("remote_id", flat=True)
+        self.installation.delete_repositories(repos_to_delete)
+
+    def update_or_create_repositories(self, repository_ids: list[int]):
+        """Update or create repositories from the given list of repository IDs."""
+        for repository_id in repository_ids:
+            try:
+                # NOTE: we save the repository ID as a string in our database,
+                # in order for PyGithub to use the API to fetch the repository by ID (not by name).
+                # it needs to be an integer, so just in case we cast it to an integer.
+                repo = self.installation_client.get_repo(int(repository_id))
+            except GithubException as e:
+                log.info(
+                    "Failed to fetch repository from GitHub",
+                    repository_id=repository_id,
+                    exc_info=True,
+                )
+                # if we lost access to the repository,
+                # we remove the repository from the database,
+                # and clean up the collaborators and relations.
+                if e.status in [404, 403]:
+                    self.installation.delete_repositories([repository_id])
+                continue
+            self._create_or_update_repository_from_gh(repo)
+
+    def _create_or_update_repository_from_gh(
+        self, gh_repo: GHRepository
+    ) -> RemoteRepository | None:
+        """
+        Create or update a remote repository from a GitHub repository object.
+
+        We also sync the collaborators of the repository with the database,
+        and create or update the organization of the repository.
+        """
+        target_id = self.installation.target_id
+        target_type = self.installation.target_type
+        # NOTE: All the repositories should be owned by the installation account.
+        # This should never happen, unless this assumption is wrong.
+        if gh_repo.owner.id != target_id or gh_repo.owner.type != target_type:
+            log.exception(
+                "Repository owner does not match the installation account",
+                repository_id=gh_repo.id,
+                repository_owner_id=gh_repo.owner.id,
+                installation_target_id=target_id,
+                installation_target_type=target_type,
+            )
+            return
+
+        remote_repo, _ = RemoteRepository.objects.get_or_create(
+            remote_id=str(gh_repo.id),
+            vcs_provider=self.vcs_provider_slug,
+        )
+
+        remote_repo.name = gh_repo.name
+        remote_repo.full_name = gh_repo.full_name
+        remote_repo.description = gh_repo.description
+        remote_repo.avatar_url = gh_repo.owner.avatar_url or self.default_user_avatar_url
```

Comment:
This seems like an odd choice? Is there only 1 owner, and putting their photo on every repo feels weird?